dynet/nodes-dropout.cc

```diff
@@ -30,7 +30,10 @@ size_t Dropout::aux_storage_size() const {
 template<class MyDevice>
 void Dropout::forward_dev_impl(const MyDevice & dev, const vector<const Tensor*>& xs, Tensor& fx) const {
   Tensor m(dim, (float*)aux_mem, fx.device, DeviceMempool::FXS);
-  TensorTools::randomize_bernoulli(m, (1.f-p), 1.f / (1.f-p));
+  std::uniform_int_distribution<> seed_dist(1, 2147483647);
```

Comment:
Also, is there a reason why you didn't just implement all of this in TensorTools? It seems like that would work as well. If there's a reason that's no problem though.

Reply:
I didn't do that partly because the TensorTools are also used for parameter initialization which I didn't want to touch, and partly because I didn't know how to properly move over the device-related code.